src/App.tsx

```diff
@@ -0,0 +1,137 @@
+import { useCallback, useMemo, useState } from 'react';
+import Button from './components/Button';
+import ColumnSelectorModal from './components/ColumnSelectorModal';
+import CountryList from './components/CountryList';
+import { createDataResource } from './data/resource';
+import type { CountryEntry, YearData } from './types/dataTypes';
+import type { ChangeEvent } from 'react';
+
+const DATA_URL = '/owid-co2-data.json';
+const resource = createDataResource(DATA_URL);
+
+const additionalColumns = ['methane', 'oil_co2', 'gas_co2', 'coal_co2', 'temperature_change_from_co2'];
+
+function App() {
+  const data = resource.read();
+
+  const countries = useMemo(() => {
+    return Object.entries(data).map(([countryName, entry]) => {
+      const country = entry as CountryEntry;
+      const name = countryName;
+      const iso = country.iso_code;
+      const years = Array.isArray(country.data) ? country.data : [];
+      const population = years[years.length - 1].population;
+      const latestPopulation = population ? population.toLocaleString() : 'N/A';
+      return { name, iso, latestPopulation, years };
```

Comment:
Consider moving this data transformation logic out of the component into a separate utility function in `utils`
It is generally a good practice to separate business logic from UI components. 
Additionally, it can be easily covered by unit tests